### DISCUSSION TITLE
MKS Robin Nano v2 Wifi Module

### DISCUSSION THREAD

**Discussion by Trvorts** [luc-github/ESP3D#565] [Questions]
Hi! Can I install ESP3D on the MKS Robin Nano V2 wifi module? Where can I find references?

**Comment by luc-github** [comment]
yes you need to compile ESP3D 3.0 changing in https://github.com/luc-github/ESP3D/blob/3.0/esp3d/configuration.h#L43
to `#define COMMUNICATION_PROTOCOL MKS_SERIAL `

to install you can use the MKS way rename the comlied FW to MksWifi.bin and copy to SD and reset board - it will automatically flash ESP board with new FW

**Comment by Trvorts** [comment]
Ok, I got it!
In this case the module only works with Marlin ported from MakerBase? Or can I use the original Marlin without LVGL UI?
Thanks for the feedback!
Cheers!

**Reply by luc-github** [reply]
I did not tested the orignal Marlin, only MKS one, which should have also some fixes for issues I have raised 
So I cannot garanty original Marlin is up to date

**Reply by Trvorts** [reply]
Ok! I'll build a bin file and test it on both fw.
Thanks again for the feedback!

**Comment by luc-github** [comment]
here doc done by MKS https://github.com/makerbase-mks/MKS-Robin-Nano-V1.X/wiki/MKS_Robin_WIFI#update-the-esp3d-firmware-to-mks-robin-wifi

**Comment by ColdBeer72** [comment]
Hi! As it's a "config after install" firmware... Please, may you share that "MksWifi.bin" recently compiled file? I have some experience with VS, but Arduino is like come back to the origins (thousand of years ago)...
Of course, it's for a MKS-WIFI module installed on a Robin Nano V3.

**Reply by luc-github** [reply]
no there are several possible configurations for esp3d so it is not really a config after install MKS support is just one of them, so sorry I do not provide any binaries or it will be never ending.

**Comment by th0m4sek** [comment]
@luc-github could you add //MKS_SERIAL Mks protocol (hardcoded to 115200 baoudrate in Mariln) in configuration? or change somehow baoudrate to 115200 when we compile MKS support automatically. It is not obvious and i lost so much time o this ;) thx for all the work.

**Reply by luc-github** [reply]
by default baudrate is 115200 in ESP3D - not sure what you mean

**Reply by th0m4sek** [reply]
i compiled raw serial first and set 250000 baudrate to match default marlin and check that all is working. later I've tried mks but baudrate was still 25000 and i was sure that there is some conflict or my module is not working, luckily i found info about 115200 in one discussion and everything is working fine, but in my opinion is not obvious.

**Reply by luc-github** [reply]
yes need documentation I agree - but I am busy coding webui right now, and I get used that, even I wrote some text, no one read documentation, so I do not put this in my top priority, but write documentation is in TODO list https://github.com/luc-github/ESP3D/tree/3.0#todo-list-to-move-to-beta in readme  that no one read :wink:

**Reply by luc-github** [reply]
@th0m4sek I have added MKs FAQ : https://github.com/luc-github/ESP3D/discussions/620
Hope it helps
